Backtracking/Generate-Permutations-Of-Vector.cpp

```diff
@@ -0,0 +1,69 @@
+#include <iostream>
+#include <cstdlib>
+
+using namespace std;
+
+int *current_solution;
+int *vector_to_permute;
```

Comment:
- Make `vector_to_permute` local variable of `main()`
- Change the type of `current_solution` and `vector_to_permute` from `int *` to `vector<int>`

Also remember to accept `vector_to_permute` as a reference (`vector<int> &vector_to_permute`) to avoid creating copies of it for each function call.

Also, if a function does not alter a variable, use the `const` modifier for it in the argument list.